code/game/objects/items/cigs_lighters.dm

```diff
@@ -321,9 +321,38 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 		return
 	var/to_smoke = smoke_all ? (reagents.total_volume * (dragtime / smoketime)) : REAGENTS_METABOLISM
 	var/mob/living/carbon/smoker = loc
-	if(!istype(smoker) || src != smoker.wear_mask)
-		reagents.remove_any(to_smoke)
-		return
+	// These checks are a bit messy but at least they're fairly readable
+	// Check if the smoker is a mob
+	if(!istype(smoker))
+		// If not, check if it's a gas mask
+		if(!istype(smoker, /obj/item/clothing/mask/gas))
+			reagents.remove_any(to_smoke)
+			return
+		// If it is, check if that mask is on a mob
+		if(!istype(smoker.loc, /mob/living/carbon))
```

Comment:
this should be `iscarbon()`. it's also kinda confusing to keep checking the loc like this and making odd assertions, you should cache the loc as a local variable in this proc, like right after line 326 just do something like

`var/atom/potential_mask = smoker.loc`

that would just make everything hopefully make more sense, especially since we access `loc` more than once throughout this proc.

Reply:
Will do

Reply:
I think I've addressed this now, but let me know if it's still unclear.